azure/types.go

```diff
@@ -96,8 +96,8 @@ type PrivateDNSLinkSpec struct {
 	LinkName          string
 }
 
-// ExtensionSpec defines the specification for a VM or VMScaleSet extension.
-type ExtensionSpec struct {
+// BootstrapingExtensionSpec defines the specification for a CAPZ Bootstrapping VM or VMSS extension.
+type BootstrapingExtensionSpec struct {
```

Comment:
Why the rename? the fields here could be reused later on for any other type of extension, nothing in here is bootstrapping specific AFAIK so I think the generic naming is appropriate

Reply:
I renamed it cause that struct is populated only by the `GetBootstrappingVMExtension()` function here and I thought it would be more clear where it's coming from. But if we want to use it more generically then we can change it back.
 https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/71a01a4acbbab2b37f8fc2010d3ceead5dadcfcc/azure/defaults.go#L361